plugin/src/main/java/org/opensearch/ml/processor/MLInferenceIngestProcessor.java

```diff
@@ -0,0 +1,368 @@
+            handler.accept(ingestDocument, null);
+        } else {
+
+            Map<String, String> modelParameters = new HashMap<>();
+
+            if (inferenceProcessorAttributes.getModelConfig() != null) {
+                modelParameters.putAll(inferenceProcessorAttributes.getModelConfig());
+            }
+            // when no input mapping is provided, default to read all fields from documents as model input
+            if (round == 0) {
+                Set<String> documentFields = ingestDocument.getSourceAndMetadata().keySet();
+                for (String field : documentFields) {
+                    getMappedModelInputFromDocuments(ingestDocument, modelParameters, field, field);
+                }
+
+            } else {
+                Map<String, String> inputMapping = processInputMap.get(i);
+                for (Map.Entry<String, String> entry : inputMapping.entrySet()) {
+                    String originalFieldName = entry.getKey();
+                    String ModelInputFieldName = entry.getValue();
+                    getMappedModelInputFromDocuments(ingestDocument, modelParameters, originalFieldName, ModelInputFieldName);
+                }
+            }
+
+            ActionRequest request = getRemoteModelInferenceRequest(modelParameters, inferenceProcessorAttributes.getModelId());
+
+            client.execute(MLPredictionTaskAction.INSTANCE, request, new ActionListener<>() {
+
+                @Override
+                public void onResponse(MLTaskResponse mlTaskResponse) {
+                    ModelTensorOutput modelTensorOutput = (ModelTensorOutput) mlTaskResponse.getOutput();
+                    if (processOutputMap == null || processOutputMap.isEmpty()) {
+                        appendFieldValue(modelTensorOutput, null, DEFAULT_OUTPUT_FIELD_NAME, ingestDocument);
+                    } else {
+                        Map<String, String> outputMapping = processOutputMap.get(i);
+
+                        for (Map.Entry<String, String> entry : outputMapping.entrySet()) {
+                            String originalModelOutputFieldName = entry.getKey(); // response
+                            String newModelOutputFieldName = entry.getValue(); // inference_result
+                            appendFieldValue(modelTensorOutput, originalModelOutputFieldName, newModelOutputFieldName, ingestDocument);
+                        }
+                    }
+                    process_predictions(ingestDocument, handler, processInputMap, processOutputMap, i + 1, round);
+                }
+
+                @Override
+                public void onFailure(Exception e) {
+                    handler.accept(null, e);
+                    return;
+                }
+            });
+        }
+
+    }
+
+    private void getMappedModelInputFromDocuments(
+        IngestDocument ingestDocument,
+        Map<String, String> modelParameters,
+        String originalFieldName,
+        String ModelInputFieldName
```

Comment:
nit: camelCase

Reply:
Nice Catch! Will change to `modelInputFieldName`